### PR TITLE
Add twirling options to SamplerV2

### LINKED ISSUE #1557: Twirling support in SamplerV2
**What is the expected feature or enhancement?**

Qiskit Runtime primitives will soon support twirling in SamplerV2. qiskit-ibm-runtime will need to make corresponding changes for that.

**Acceptance criteria**

- [ ] Support twirling options for SamplerV2
- [ ] Documentation should indicate twirling is only done on gates/measurements that are not used in a control flow

### PR DESCRIPTION
**Pull request by ptristan3**
### Details and comments

Fixes #1557

### CODE REVIEW

File: release-notes/unreleased/1557.feat.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+The Sampler V2 starts to support twirling.
```
Comment (kt474) [inline release-notes/unreleased/1557.feat.rst:1]:
```suggestion
`SamplerV2` now supports twirling.
```

**Review by kt474 — APPROVED** [review]
Minor comment but LGTM

**Review by ihincks — COMMENTED** [review]
Thanks @ptristan3. It looks like we'll need to reword some of the `TwirlingOptions` docstrings to make them more generic, as they are specific to the estimator right now.

We should also mention that measurement twirling will only be applied to those measurement registers not involved in conditional logic.

File: qiskit_ibm_runtime/options/twirling_options.py [hunk 2]
```diff
@@ -33,7 +33,7 @@ class TwirlingOptions:
         enable_gates: Whether to apply 2-qubit gate twirling. Default: False.
 
         enable_measure: Whether to enable twirling of expectation value measurements
-          in Estimator. Default: True.
+          in the Primitive. Default: True.
```
Comment (jyu00) [inline qiskit_ibm_runtime/options/twirling_options.py:36]: The wording should just say measurements, and the sentence on conditional should also be here, e.g. 

`
enable_measure: Whether to enable twirling of measurements. The twirling can only be applied to those measurement registers not involved within a conditional logic. Default: True.
`

**Review by jyu00 — COMMENTED** [review]
I don't think measurement twirling for sampler is enabled on the server side yet. We should wait for that before merging.

**Comment by jyu00** [comment]
@ptristan3 can we also add twirling to `test_program_inputs` in `TestSamplerOptions`? Whether the option gets passed to the server correctly is very important.

File: qiskit_ibm_runtime/options/twirling_options.py [hunk 3]
```diff
@@ -32,8 +32,8 @@ class TwirlingOptions:
     Args:
         enable_gates: Whether to apply 2-qubit gate twirling. Default: False.
 
-        enable_measure: Whether to enable twirling of expectation value measurements
-          in Estimator. Default: True.
+        enable_measure: Whether to enable twirling of measurements. The twirling can only be applied to
```
Comment (ihincks) [inline qiskit_ibm_runtime/options/twirling_options.py:35]: slight clarification:

```suggestion
        enable_measure: Whether to enable twirling of measurements. Twirling will only be applied to
```

File: release-notes/unreleased/1557.feat.rst [hunk 4]
```diff
@@ -0,0 +1,2 @@
+`SamplerV2` now supports twirling.
+The twirling can only be applied to those measurement registers not involved within a conditional logic.
```
[the remark this answers is not in the document]
Comment (ptristan3) [reply release-notes/unreleased/1557.feat.rst]: thank you @ihincks!

**Pull request merged**